### PR TITLE
Fix 43612

### PR DESCRIPTION
**Pull request by artem-fedorov**
Hello,

I tried to send PUT request with a few Parameters, but I don't see them in my Request. I found bug for it https://bz.apache.org/bugzilla/show_bug.cgi?id=43612 

If I send PUT request with RawBody it works fine. 

Please, review my PR with fix

### CODE REVIEW

File: src/protocol/http/org/apache/jmeter/protocol/http/sampler/HTTPHC4Impl.java [hunk 1]
```diff
@@ -1649,6 +1648,43 @@ else if(getSendParameterValuesAsPostBody()) {
             }
             StringEntity requestEntity = new StringEntity(entityBodyContent.toString(), charset);
             entity.setEntity(requestEntity);
+        } else if (hasArguments()) {
```
Comment (undera) [inline src/protocol/http/org/apache/jmeter/protocol/http/sampler/HTTPHC4Impl.java]: Is it possible to extract this long code piece into separate method? For better readability.

**Comment by artem-fedorov** [comment]
something wrong with TCP_TESTS

File: src/protocol/http/org/apache/jmeter/protocol/http/sampler/HTTPHC4Impl.java (hunk 1 above)
Comment (artem-fedorov) [reply src/protocol/http/org/apache/jmeter/protocol/http/sampler/HTTPHC4Impl.java]: Done

**Comment by pmouawad** [comment]
Hello @artem-fedorov ,
Is your PR for the bug you indicate which is supposed to be fixed long time ago or for this one:

- https://bz.apache.org/bugzilla/show_bug.cgi?id=60782

See those discussions:

- http://mail-archives.apache.org/mod_mbox/jmeter-dev/201702.mbox/%3CCAH9fUpZ6dKz4f4DFuVU56qc5cC0PkfCT3Qxq1r38_UbNmsiruA%40mail.gmail.com%3E
- http://mail-archives.apache.org/mod_mbox/jmeter-dev/201702.mbox/%3CCAH9fUpbSfK95m2ZYQwRJqEN%3D4NBmfSzqY3-FqNpSc2xz2hDD%2BA%40mail.gmail.com%3E


Regards

**Comment by artem-fedorov** [comment]
@pmouawad 
I think it fix for old bug 43612.
See screenshots:
https://bz.apache.org/bugzilla/attachment.cgi?id=35796
https://bz.apache.org/bugzilla/attachment.cgi?id=35797

**Pull request closed without merging**